src/PSR7/SpecFinder.php

```diff
@@ -215,7 +231,7 @@ public function findResponseSpec(ResponseAddress $addr) : ResponseSpec
     public function findHeaderSpecs(OperationAddress $addr) : array
     {
         // Response headers are specified differently from request headers
-        if ($addr instanceof ResponseAddress) {
+        if ($addr instanceof ResponseAddress || $addr instanceof CallbackResponseAddress) {
```

Comment:
Not sure if it worth it, but we can try to itroduce an interface or something like this, instead of doing additional manual `assert`s  and `instanceof` checks and dropping the signatures

Reply:
I thought about the same, but then decided against it, as there will not be that many other addresses (or right now there isn't any?). Furthermore, as you said in the summary comment, this would be a bigger refactoring, so a further reason I didn't do that yet.